src/hotspot/share/compiler/compilerDirectives.hpp

```diff
@@ -185,6 +185,39 @@ class ControlIntrinsicIter {
   ControlIntrinsicIter& operator++();
 };
 
+class ControlIntrinsicValidator {
+ private:
+  bool _valid;
+  char* _bad;
+
+ public:
+  ControlIntrinsicValidator(ccstrlist option, bool disabled_all) : _valid(true), _bad(nullptr) {
```

Comment:
The argument of `ControlIntrinsicIter` is called `disable_all`, so it should be `disable_all` here as well.

The comment in `ControlIntrinsicIter` is confusing:
```
// if disable_all is true, it accepts DisableIntrinsic(deprecated) and all intrinsics
// appear in the list are to disable
```

Could you explain what exactly `disabled_all` is needed for?

Reply:
before ControlIntrinsic, we have only this option: 
-XX:DisableIntrinsic=_id1,_id2,_id3,...

With ControlIntrinsic, we can have this form, + means enable, - means disable. 
-XX:ControlIntrinsic=+_id1, -_id2,+_id3,...

After DisableIntrinsic, we can still use this form. in this case, ControlIntrinsic is just an alias of  DisableIntrinsic. 
-XX:ControlIntrinsic=_id1, _id2,_id3,...